include/exadg/time_integration/time_int_abm_base.h

```diff
@@ -0,0 +1,270 @@
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_EXADG_TIME_INTEGRATION_TIME_INT_ABM_BASE_H_
+#define INCLUDE_EXADG_TIME_INTEGRATION_TIME_INT_ABM_BASE_H_
+
+
+#include <exadg/time_integration/ab_constants.h>
+#include <exadg/time_integration/am_constants.h>
+#include <exadg/time_integration/push_back_vectors.h>
+#include <exadg/time_integration/time_int_multistep_base.h>
+
+namespace ExaDG
+{
+/**
+ * This class implements the purely explicit Adams--Bashforth--Moulton predictor corrector method.
+ */
+template<typename Operator, typename VectorType>
+class TimeIntAdamsBashforthMoultonBase : public TimeIntMultistepBase
+{
+  using Number = typename VectorType::value_type;
+
+public:
+  TimeIntAdamsBashforthMoultonBase(std::shared_ptr<Operator> pde_operator_in,
+                                   double const              start_time_,
+                                   double const              end_time_,
+                                   unsigned int const        max_number_of_time_steps_,
+                                   unsigned int const        order_,
+                                   bool const                start_with_low_order_,
+                                   bool const                adaptive_time_stepping_,
+                                   RestartData const &       restart_data_,
+                                   MPI_Comm const &          mpi_comm_,
+                                   bool const                is_test_)
+    : TimeIntMultistepBase(start_time_,
+                           end_time_,
+                           max_number_of_time_steps_,
+                           order_,
+                           start_with_low_order_,
+                           adaptive_time_stepping_,
+                           restart_data_,
+                           mpi_comm_,
+                           is_test_),
+      pde_operator(pde_operator_in),
+      // order of predictor can be chosen one below order of corrector
+      ab(order_ - 1, start_with_low_order_),
+      am(order_, start_with_low_order_),
+      vec_evaluated_operators(order_ - 1)
+  {
+    AssertThrow(order_ >= 1,
+                dealii::ExcMessage("Oder of ABM time integrator has to be at least 1."));
+  }
+
+  void
+  print_iterations() const
+  {
+    // explicit time integration -> one iteration
+    print_list_of_iterations(pcout, {"Adams-Bashforth-Moulton"}, {1});
```

Comment:
If I remember correctly, we do it somewhat differently in other parts of ExaDG (Explicit == 0 iterations), see e.g. the dual-splitting scheme: https://github.com/exadg/exadg/blob/56272c15cceb9ecc1300a176c7c5f11cb78604ca/include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_dual_splitting.cpp#L954

Reply:
I changed it to zero iterations :)